internal/repository/evmreader.go

```diff
@@ -57,21 +59,22 @@ func (pg *Database) InsertInputsAndUpdateLastProcessedBlock(
 			"rawData":     input.RawData,
 			"blockNumber": input.BlockNumber,
 			"appAddress":  input.AppAddress,
+			"epochId":     input.EpochId,
 		}
 		_, err = tx.Exec(ctx, query, inputArgs)
 		if err != nil {
-			return fmt.Errorf("%w: %w", errInsertInputs, err)
+			return errors.Join(fmt.Errorf("%w: %w", errInsertInputs, err), tx.Rollback(ctx))
```

Comment:
You can remove the `fmt.Errorf` call by doing `errors.Join(errInsertInputs, err, tx.Rollback(ctx))`.

Reply:
Done